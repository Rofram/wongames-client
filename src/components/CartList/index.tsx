import Button from 'components/Button'
import Empty from 'components/Empty'
import GameItem, { GameItemProps } from 'components/GameItem'
import Link from 'next/link'
import * as S from './styles'

export type CartListProps = {
  items?: GameItemProps[]
  total?: string
  hasButton?: boolean
}

const CartList = ({ items = [], total, hasButton = false }: CartListProps) => (
  <S.Wrapper isEmpty={!items.length}>
    {items.length ? (
      <>
        {items.map((item, index) => (
          <GameItem key={`${item.title}-${index}`} {...item} />
        ))}

        <S.Footer>
          {hasButton ? (
            <>
              <S.Total>{total}</S.Total>{' '}
              <Link href="/cart" passHref>
                <Button aria-label="Buy it now" as="a">
                  Buy it now
                </Button>
              </Link>
            </>
          ) : (
            <>
              <span>Total</span> <S.Total>{total}</S.Total>
            </>
          )}
        </S.Footer>
      </>
    ) : (
      <Empty
        title="Your cart is Empty"
        description="Go back to the store and explore great games and offers"
        color="black"
        hasLink
      />
    )}
  </S.Wrapper>
)

export default CartList
