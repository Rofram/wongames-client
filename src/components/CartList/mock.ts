import { CartListProps } from '.'

import gameItemMock, { paymentInfoMock } from 'components/GameItem/mock'

const mock: Required<CartListProps> = {
  items: Array.from({ length: 4 }, () => gameItemMock),
  total: '$250,00',
  hasButton: false
}

export const withPaymentInfo: CartListProps = {
  items: Array.from({ length: 4 }, () => ({
    ...gameItemMock,
    paymentInfo: paymentInfoMock
  })),
  total: '$300,00'
}

export default mock
