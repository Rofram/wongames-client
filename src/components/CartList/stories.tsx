import { Story, Meta } from '@storybook/react'
import CartList, { CartListProps } from '.'

import cartListMock from './mock'

export default {
  title: 'CartList',
  component: CartList,
  args: cartListMock,
  parameters: {
    backgrounds: {
      default: 'won-dark'
    }
  }
} as Meta

export const Default: Story<CartListProps> = (args) => (
  <div style={{ maxWidth: 800, margin: '0 auto' }}>
    <CartList {...args} />
  </div>
)
