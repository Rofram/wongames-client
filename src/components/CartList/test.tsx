import { screen } from '@testing-library/react'
import theme from 'styles/theme'
import { renderWithTheme } from 'utils/tests/helpers'

import CartList from '.'

import props from './mock'

jest.mock('components/GameItem', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="GameItem" />
    }
  }
})

jest.mock('components/Empty', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Empty" />
    }
  }
})

describe('<CardList />', () => {
  it('should render the card list', () => {
    const { container } = renderWithTheme(<CartList {...props} />)

    expect(screen.getAllByTestId('GameItem')).toHaveLength(props.items.length)

    expect(screen.getByText('Total')).toBeInTheDocument()

    expect(screen.getByText(props.total)).toHaveStyle({
      color: theme.colors.primary
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the button', () => {
    renderWithTheme(<CartList {...props} hasButton />)

    expect(screen.getByLabelText(/buy it now/i)).toBeInTheDocument()
  })

  it('should render <Empty /> with no items in cart', () => {
    renderWithTheme(<CartList {...props} items={[]} />)

    expect(screen.getByTestId('Empty')).toBeInTheDocument()

    expect(screen.queryByText(/total/i)).not.toBeInTheDocument()
  })
})
