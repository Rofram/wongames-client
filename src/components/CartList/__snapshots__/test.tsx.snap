// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> should render the card list 1`] = `
.c0 {
  background: #FAFAFA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.c1 {
  background: #eee;
  color: #030517;
  font-weight: 600;
  font-size: 1.4rem;
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  color: #F231A5;
}

@media (min-width:768px) {
  .c1 {
    font-size: 1.6rem;
    padding: 2.4rem;
  }
}

<div
  class="c0"
>
  <div
    data-testid="GameItem"
  />
  <div
    data-testid="GameItem"
  />
  <div
    data-testid="GameItem"
  />
  <div
    data-testid="GameItem"
  />
  <div
    class="c1"
  >
    <span>
      Total
    </span>
     
    <span
      class="c2"
    >
      $250,00
    </span>
  </div>
</div>
`
